### PR TITLE
IGAPP-675: Add types to web-ts for NoImplicitAny rule

### PR DESCRIPTION
**Pull request by Znippet**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

Added missing types to method parameters and variables. Until now there are issues with untyped imports, so I didn't apply the rule itself

### CODE REVIEW

File: web-ts/src/routes/__tests__/PoisPage.spec.tsx [hunk 1]
```diff
@@ -84,11 +89,13 @@ describe('PoisPage', () => {
...
-    mockUseLoadFromEndpointOnce(() => ({
+    const useLoadFromEndpointMock = (() => ({
       data: pois,
       loading: false,
-      error: null
-    }))
+      error: null,
+      refresh: () => null
+    })) as typeof useLoadFromEndpoint
+    mockUseLoadFromEndpointOnce(useLoadFromEndpointMock)
```
Comment (klinzo) [inline web-ts/src/routes/__tests__/PoisPage.spec.tsx:98]: Maybe we could move this to a helper method/create a global mock for this.

**Review by klinzo — APPROVED** [review]

**Review by maxammann — APPROVED** [review]

File: web-ts/src/routes/__tests__/PoisPage.spec.tsx (hunk 1 above)
Comment (Znippet) [reply web-ts/src/routes/__tests__/PoisPage.spec.tsx]: done

File: web-ts/src/testing/mockUseLoadFromEndpoint.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { useLoadFromEndpoint } from 'api-client'
```
Comment (klinzo) [inline web-ts/src/testing/mockUseLoadFromEndpoint.tsx:1]: I think this makes more sense in the api-client as `useLoadFromEndpoint` is also imported from there.

File: web-ts/src/testing/mockUseLoadFromEndpoint.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+const mockData = (data): typeof useLoadFromEndpoint => {
+  return (() => ({
+    data: data,
+    loading: false,
+    error: null,
+    refresh: () => null
+  })) as typeof useLoadFromEndpoint
+}
+
+export function mockUseLoadFromEndpointWitData<T>(data: T): void {
```
Comment (klinzo) [inline web-ts/src/testing/mockUseLoadFromEndpoint.tsx:13]: Please use arrow funcs

**Pull request merged**